components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -242,11 +247,24 @@ public void removeExpiredSessionData(Timestamp timestamp) {
             statement = connection.prepareStatement(sqlDeleteTask);
             statement.setTimestamp(1, timestamp);
             statement.execute();
-            connection.commit();
-        } catch (SQLException | IdentityException e) {
-            log.error("Error while removing Session Data.", e);
+            if (!connection.getAutoCommit()) {
+                connection.commit();
+            }
+        } catch (SQLException e) {
+            log.error("Error while removing Session Data ", e);
+        } catch (IdentityException e) {
+            log.error("Error while removing Session Data", e);
         } finally {
-            IdentityDatabaseUtil.closeAllConnections(connection, null, statement);
+            try {
+                if (statement != null) {
```

Comment:
I'm not quite familiar with the internals of the aforesaid method. You might have to double check on the method imply and see if it properly handles what's been pointed out and if it does, adopt it into what you've implementation.